src/charm.py

```diff
@@ -303,17 +300,43 @@ def _current_mimir_config(self) -> dict:
             )
             return {}
 
-    @property
-    def _mimir_version(self) -> Optional[str]:
-        if not self._container.can_connect():
-            return None
+    def _set_alerts(self) -> None:
+        """(Re-)create alert rule files for all Mimir consumers.
 
-        version_output, _ = self._container.exec(["/bin/mimir", "-version"]).wait_output()
-        # Output looks like this:
-        # Mimir, version 2.4.0 (branch: HEAD, revision: 32137ee)
-        if result := re.search(r"[Vv]ersion:?\s*(\S+)", version_output):
-            return result.group(1)
-        return None
+        Returns: True if alerts rules have changed, otherwise False.
+        Raises: BlockedStatusError exception if PebbleError, ProtocolError, PathError exceptions
+            are raised by container.remove_path
+        """
+        try:
+            self._container.remove_path(RULES_DIR, recursive=True)
+        except PebbleError as e:
+            logger.error("Failed to remove alerts directory: %s", e)
+            raise BlockedStatusError("Failed to remove alerts directory; see debug logs")
```

Comment:
I've made it a habit to always `raise X from e` whenever I'm wrapping errors, so as not to lose any trace info. Maybe you're catching this event already someplace else, but I think it'd be good practice anyway.

Reply:
Agreed. This is just a re-shuffle. I'll follow back through for cleanup.